explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/shap/ShapKernelExplainer.java

```diff
@@ -0,0 +1,660 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.explainability.local.shap;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.commons.math3.exception.MathArithmeticException;
+import org.apache.commons.math3.util.CombinatoricsUtils;
+import org.kie.kogito.explainability.local.LocalExplainer;
+import org.kie.kogito.explainability.model.Prediction;
+import org.kie.kogito.explainability.model.PredictionInput;
+import org.kie.kogito.explainability.model.PredictionOutput;
+import org.kie.kogito.explainability.model.PredictionProvider;
+import org.kie.kogito.explainability.model.ShapPrediction;
+import org.kie.kogito.explainability.utils.MatrixUtils;
+import org.kie.kogito.explainability.utils.RandomChoice;
+import org.kie.kogito.explainability.utils.WeightedLinearRegression;
+import org.kie.kogito.explainability.utils.WeightedLinearRegressionResults;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the SHAP kernel explainer as per
+ * https://proceedings.neurips.cc/paper/2017/file/8a20a8621978632d76c43dfd28b67767-Paper.pdf
+ * see also https://github.com/slundberg/shap/blob/master/shap/explainers/_kernel.py
+ */
+public class ShapKernelExplainer implements LocalExplainer<double[][]> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShapKernelExplainer.class);
+
+    public ShapKernelExplainer() {
+        // nothing to set
+    }
+
+    private ShapDataCarrier initialize(ShapPrediction prediction, PredictionProvider model) {
+        ShapConfig config = prediction.getConfig();
+
+        // get shapes of input and output data
+        int[] shape = MatrixUtils.getShape(config.getBackgroundMatrix());
+        int rows = shape[0];
+        int cols = shape[1];
+
+        if (rows > 100) {
+            LOGGER.debug("Warning: Background data sets larger than 100 samples might be slow!");
+        }
+
+        // establish background data
+        CompletableFuture<List<PredictionOutput>> modelOut = model.predictAsync(config.getBackground());
+        CompletableFuture<double[][]> modelNull = modelOut.thenApply(MatrixUtils::matrixFromPredictionOutput);
```

Comment:
As far as I can see `modelOut` is only used here so you should be able to chain the calls
```suggestion
        CompletableFuture<double[][]> modelNull = model.predictAsync(config.getBackground())
                .thenApply(MatrixUtils::matrixFromPredictionOutput);
```